### ISSUE TITLE
i need help asap plssssssss

### ISSUE THREAD

**Issue by faridamousa** [CV-Wang/RapeNet#2]
i have multiple questions:

1) are the images supposed to be resized when i run the preprocess_RFCP.py and preprocess_RFRB.py
2) i get multiple warnings before the epochs start: RuntimeWarning: invalid value encountered in scalar divide slope = ssxym / ssxm
RuntimeWarning: invalid value encountered in sqrt t = r * np.sqrt(df / ((1.0 - r + TINY)*(1.0 + r + TINY)))
RuntimeWarning: invalid value encountered in scalar divide slope_stderr = np.sqrt((1 - r**2) * ssym / ssxm / df)
RuntimeWarning: Mean of empty slice. return _methods._mean(a, axis=axis, dtype=dtype,
RuntimeWarning: invalid value encountered in divide ret = ret.dtype.type(ret / rcount)
how can i fix them? i have been trying but nothing is working
3) when i train the model, the Loss, MAE, MSE values are really high and R^2 is really low.
4) the inferred values are so so low, like less than half the manual count when i test
PLEASE HELP
I AM IN DESPERATE NEED OF HELP

**Comment by faridamousa** [comment]
PLEASE HELPPPP

**Comment by faridamousa** [comment]
HELP PLEASEEEEEE

**Comment by faridamousa** [comment]
the results are not the same as the paper

**Comment by CV-Wang** [comment]
Hello, I have been busy with work recently and only saw my email today. Have you tried the Google Cloud Drive model I provided for testing? If you have tried, please share a screenshot with me for a look. Thank you.







---- Replied Message ----
| From | ***@***.***> |
| Date | 5/9/2024 06:43 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | Re: [CV-Wang/RapeNet] i need help asap plssssssss (Issue #2) |

the results are not the same as the paper

**Comment by faridamousa** [comment]
i want to train the model myself

**Comment by faridamousa** [comment]
when i train the loss, mae, mse values keep going up and down and mse values are the worst 
and in val epoch the mse, mae become bigger than 300 and r2 is very low

**Comment by faridamousa** [comment]
hello, 
when i run preprocess_RFRB and preprocess_RFCP should images's sizes change? because when i do they stay the same.
i want to train the model myself and
when i train the loss, mae, mse values keep going up and down and mse values are the worst
and in val epoch the mse, mae become bigger than 300 and r2 is very low
will send u screenshots of the first 60 epochs
even if some r2 are somewhat good, they are not saved in the model path to be tested

thank you
________________________________________
From: CV-Wang ***@***.***
Sent: Thursday, May 09, 2024 3:57 AM
To: CV-Wang/RapeNet
Cc: Farida Hisham Saeed Mohamed Mousa; Author
Subject: [EXTERNAL EMAIL] - Re: [CV-Wang/RapeNet] i need help asap plssssssss (Issue #2)

Hello, I have been busy with work recently and only saw my email today. Have you tried the Google Cloud Drive model I provided for testing? If you have tried, please share a screenshot with me for a look. Thank you.







---- Replied Message ----
| From | ***@***.***> |
| Date | 5/9/2024 06:43 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | Re: [CV-Wang/RapeNet] i need help asap plssssssss (Issue #2) |

the results are not the same as the paper

**Comment by faridamousa** [comment]
it doesn't make sense that when i train the model on the dataset provided, the result aren't even close to the results written on the paper or even the results of the pretrained model.
pretrained model results and results in paper did not come out the same with me

**Comment by CV-Wang** [comment]
Can the heat intensity of your test results be aligned?

**Comment by faridamousa** [comment]
i don't understand
what do you mean by heat intensity?
please read the comments i wrote and the email i sent you.

**Comment by CV-Wang** [comment]
This was my work more than a year ago, and I am no longer doing research in this direction. Because I'm not sure whether the problem you are encountering is caused by the data or the code part. I did a data check and found that some RFRB images uploaded last year were missing and have been added. I just tried it and the test results are correct. Or you can share your experimental results with me to determine your problem.
![YT21](https://github.com/CV-Wang/RapeNet/assets/129062567/b745fdd7-c9e2-4657-8c6f-0f3ee722c4aa)

**Comment by faridamousa** [comment]
hi
when i run the preprocess_RFRB and preprocess_RFCP before training, nothing happens to the image in terms of size.
should the size change?
and when i start training the values are not good at all. i think it's code problem or maybe the images?
and when i test, the results aren't good because the training wasn't good.
________________________________________
From: CV-Wang ***@***.***
Sent: Sunday, May 12, 2024 9:28 AM
To: CV-Wang/RapeNet
Cc: Farida Hisham Saeed Mohamed Mousa; Author
Subject: [EXTERNAL EMAIL] - Re: [CV-Wang/RapeNet] i need help asap plssssssss (Issue #2)

This was my work more than a year ago, and I am no longer doing research in this direction. Because I'm not sure whether the problem you are encountering is caused by the data or the code part. I did a data check and found that some RFRB images uploaded last year were missing and have been added. I just tried it and the test results are correct. Or you can share your experimental results with me to determine your problem.
YT21.png (view on web)<https://github.com/CV-Wang/RapeNet/assets/129062567/b745fdd7-c9e2-4657-8c6f-0f3ee722c4aa>

**Comment by CV-Wang** [comment]
The function of preprocess.py is to convert rectangular boxes or center points labeled in different formats into unified center point data and save them as .npy files to facilitate training. You can refer to the definitions of mae, mse, rmae, rmse, rrmse, and R2 in the code. I checked the information during my research and found that there may be different definitions in different articles. You can check whether the loss converges during training. If it does not converge, it is most likely a data problem.

**Comment by faridamousa** [comment]
the loss stays within teens and twenties 
my problem is with the mse, it's always thirties or above and when val epoch comes rmse, rmae are bigger than 300 and r2 is very low
what should i do?
how can it be a data problem?
________________________________________
From: CV-Wang ***@***.***
Sent: Monday, May 13, 2024 5:10 AM
To: CV-Wang/RapeNet
Cc: Farida Hisham Saeed Mohamed Mousa; Author
Subject: [EXTERNAL EMAIL] - Re: [CV-Wang/RapeNet] i need help asap plssssssss (Issue #2)

The function of preprocess.py is to convert rectangular boxes or center points labeled in different formats into unified center point data and save them as .npy files to facilitate training. You can refer to the definitions of mae, mse, rmae, rmse, rrmse, and R2 in the code. I checked the information during my research and found that there may be different definitions in different articles. You can check whether the loss converges during training. If it does not converge, it is most likely a data problem.

**Comment by CV-Wang** [comment]
You can try the test results of the model you trained yourself. Try both RapeNet and RapeNet+ to see if the difference in estimated numbers is too large, and then consider adjusting the details of the training parameters (epoch, learning rate, etc.)
